src/commands/dev/index.js

```diff
@@ -132,7 +132,12 @@ function initializeProxy(port, distDir, projectDir) {
     if (!isEmpty(pathHeaderRules)) {
       Object.entries(pathHeaderRules).forEach(([key, val]) => res.setHeader(key, val))
     }
-    res.writeHead(req.proxyOptions.status || proxyRes.statusCode, proxyRes.headers)
+    // leave 3xx redirects handling to browser
+    if (proxyRes.statusCode >= 300 && proxyRes.statusCode < 400) {
```

Comment:
In fact, I'm considering we probably should always return the proxyRes.statusCode, no matter it is 3xx or 4xx or 5xx.

Reply:
Thank you for opening this PR @bencao. The reason we have `req.proxyOptions.status` is to force a certain status code, for example, `200!` or `404!`

Reply:
configuration like this is a bit vague, what if the api server returns http 500?

```
[[redirects]]
  from = "/search"
  to = "https://api.mysearch.com"
  status = 200
  force = true
```

Reply:
as a user, my actual intention is basically requesting a server side transparent proxy, not enforcing the status code to 200

Reply:
I think we would still enforce `200` as status

Reply:
ok.. do you have some ideas about how we can gracefully solve the above proxy server 304 resulted in blank page issue?

Reply:
Can you try out this branch https://github.com/netlify/cli/pull/843 and see if that fixes the problem for you!

Reply:
I checked the branch, it seems to be a bit too complicated to solve my very specific problem..